src/Streamly/Internal/Wrapper.hs

```diff
@@ -0,0 +1,23 @@
+-- |
+-- Module      : Streamly.Internal.Wrapper
+-- License     : BSD3
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+-- Wrapper functions for internal packages.
+--
+module Streamly.Internal.Wrapper
+    ( sizeOfElem
+    , sizeOfPrimElem
+    )
+where
+
+import qualified Foreign.Storable  as ST
+import Data.Primitive.Types as Prim (Prim, sizeOf )
+
+sizeOfElem :: forall a . (ST.Storable a) => a -> Int
+sizeOfElem _ = max 1 $ ST.sizeOf (undefined :: a)
```

Comment:
We can define this in Data.Array.Foreign.Mut.Type. This is array specific and not a general API.